### PR TITLE
Duplicate check

### PR DESCRIPTION
**Pull request by supersecretcatia**
- Check for duplicate clientId and positionId and generate another one
- Check for duplicate username and throw error

### CODE REVIEW

File: src/main/java/com/iggroup/universityworkshopmw/domain/services/ClientService.java [hunk 1]
```diff
@@ -50,4 +55,20 @@ Client getClientDataFromMap(String clientId) throws NoAvailableDataException {
...
+   private String checkIfDuplicateClientId() {
+      String uniqueId;
+      do {
+         uniqueId = Helper.createUniqueId(ID_PREFIX);
+      } while (clientIdToClientModelMap.containsKey(uniqueId));
+      return uniqueId;
+   }
+
+   private void checkIfDuplicateUsername(Client client) throws DuplicatedDataException {
+      Optional<Client> duplicatedUsername = clientIdToClientModelMap.values().stream().filter(c -> c.getUserName().equals(client.getUserName())).findFirst();
```
Comment (supersecretliam) [inline src/main/java/com/iggroup/universityworkshopmw/domain/services/ClientService.java]: could you separate the stream onto multiple lines?

File: src/main/java/com/iggroup/universityworkshopmw/integration/controllers/ClientController.java [hunk 2]
```diff
@@ -54,6 +56,11 @@
          ClientDto responseBody = ClientDtoTransformer.transform(clientService.storeNewClient(clientTransformed));
          return new ResponseEntity<>(responseBody, OK);
 
+      } catch (DuplicatedDataException e) {
+         String userName = clientDto.getUserName();
+         log.info("Duplicated username={}, exceptionMessage={}", userName, e);
+         return new ResponseEntity<>("Username=" + userName + " is already used. Please create another one", HttpStatus.BAD_REQUEST);
```
Comment (supersecretliam) [inline src/main/java/com/iggroup/universityworkshopmw/integration/controllers/ClientController.java]: static import?

**Review by supersecretliam — COMMENTED** [review]
Looks good. Only a couple of small things. I tried to think of another way to do the `do while` loops, but I couldn't think of a better way :)

File: src/main/java/com/iggroup/universityworkshopmw/domain/services/ClientService.java (hunk 1 above)
Comment (supersecretcatia) [reply src/main/java/com/iggroup/universityworkshopmw/domain/services/ClientService.java]: yup will do!

File: src/main/java/com/iggroup/universityworkshopmw/integration/controllers/ClientController.java (hunk 2 above)
Comment (supersecretcatia) [reply src/main/java/com/iggroup/universityworkshopmw/integration/controllers/ClientController.java]: yup!

**Review by supersecretliam — APPROVED** [review]

**Pull request merged**